### DISCUSSION TITLE
Infer schema for NDJSON

### DISCUSSION THREAD

**Discussion by Smotrov** [apache/datafusion#10900] [Q&A] [answered]
I have a data in NDJSON format. 
The thing is some fields there are optional and populated when needed. 
As I understand, DF is using the first row only by default to infer schema. Than when it see that certain fields are missing in som of the next rows it shows an error `Fail to merge schema field 'my_fieldname' because the from data_type ... does not equal ...`. 

So is it possible to specify N rows which should be used to infer schema?

**Comment by jonahgao** [comment]
Could you provide a small reproducible example? I’m not familiar with this area, but I may be able to help with debugging.

**Comment by Jefffrey** [comment, marked as answer]
Should be able to use `schema_infer_max_records` from `NdJsonReadOptions` to specify how many rows to use for inference now:

https://github.com/apache/datafusion/blob/d24eb4a23156b7814836e765d5890186ab40682f/datafusion/core/src/datasource/file_format/options.rs#L445-L468

(https://github.com/apache/datafusion/pull/18920 adds a nicer API to configure this setting)
